be/dtm0_log.h

```diff
@@ -0,0 +1,522 @@
+ * @param log Pointer to a m0_be_dtm0_log that has to be initialized.
+ * @param cs Pointer to a clock source. This will be used for comparisons
+ *        involving log record timestamps.
+ * @param isvstore A flag to indicate whether the dtm0 log that we are trying
+ *        to init is a volatile of persistent store.
+ *
+ * @return 0 on success. Anything else is a failure.
+ */
+M0_INTERNAL int m0_be_dtm0_log_init(struct m0_be_dtm0_log  *log,
+                                    struct m0_dtm0_clk_src *cs,
+                                    bool                    is_plog);
+
+/**
+ * Finalize a dtm0 log. We need to call this routine once after we have
+ * finished performing all operations on it.
+ *
+ * @pre m0_be_dtm0_log__invariant needs to be satisfied.
+ * @post All fields of *log are finalized. For volatile log the memory
+ *       allocated by m0_be_dtm0_log_alloc is freed.
+ *
+ * @param log Pointer to a valid log.
+ *
+ * @return None
+ */
+M0_INTERNAL void m0_be_dtm0_log_fini(struct m0_be_dtm0_log **log);
+
+/**
+ * Free the memory allocated by m0_be_dtm0_log_alloc
+ *
+ * @pre m0_be_dtm0_log->dl_is_persistent needs to be false.
+ * @post *log is set to NULL.
+ *
+ * @param log Pointer to a log structure that has been previously allocated.
+ *
+ * @return None
+ */
+M0_INTERNAL void m0_be_dtm0_log_free(struct m0_be_dtm0_log **log);
+
+/**
+ * For performing an operation on a persistent log, we need to take the
+ * appropriate number of be transaction credit.
+ *
+ * @pre None
+ * @post None
+ *
+ * @param op Operation code for the type of dtm0 log operation that we want
+ *        to perform. @see @ref m0_be_dtm0_log_credit_op.
+ * @param txd A valid dtm0 tx descriptor that we want to store in the log
+ *        record.
+ * @param payload The payload is an opaque structure for dtm0 log to store
+ *        in the log record.
+ * @param seg The be segment on which the log resides.
+ * @param A pointer to a valid m0_dtm0_log_rec.
+ * @param accum This contains the number of credits required for performing
+ *        the operation.
+ * @return None
+ */
+M0_INTERNAL void m0_be_dtm0_log_credit(enum m0_be_dtm0_log_credit_op op,
+                                       struct m0_dtm0_tx_desc	    *txd,
+                                       struct m0_buf		    *payload,
+                                       struct m0_be_seg             *seg,
+                                       struct m0_dtm0_log_rec	    *rec,
+                                       struct m0_be_tx_credit       *accum);
+
+/**
+ * This routine is used to create a dtm0 persistent log.
+ *
+ * @pre tx != NULL
+ * @pre seg != NULL
+ * @post None
+ *
+ * @param tx handle to an already opened transaction.
+ * @param seg be segment on which to create the log.
+ * @param out variable in which to store the pointer to the newly created
+ *        log.
+ * @return 0 on success, anything else is a failure.
+ */
+M0_INTERNAL int m0_be_dtm0_log_create(struct m0_be_tx        *tx,
```

Comment:
So, the m0_be_dtm0_log_create() is used to create a persistent log,
and the m0_be_dtm0_log_alloc() is used to create a volatile log?

Reply:
Yes. That is correct!